File: lambdas/node_js/http_request.js
import https from 'https';

// https://api.monobank.ua/bank/currency

export const handler = async (event) => {
    const options = {
        hostname: 'api.monobank.ua',
        path: '/bank/currency',
        method: 'GET',
    };

    return new Promise((resolve, reject) => {
        const req = https.request(options, (res) => {
            let data = '';

            // Listen for data chunks
            res.on('data', (chunk) => {
                data += chunk;
            });

            // Response has ended
            res.on('end', () => {
                resolve({
                    statusCode: 200,
                    body: data,
                });
            });
        });

        // Handle request errors
        req.on('error', (error) => {
            console.error('Error making API request:', error);
            resolve({
                statusCode: 500,
                body: JSON.stringify({ message: 'Internal Server Error' }),
            });
        });

        req.end();
    });
};

File: lambdas/python/http_request.py
import urllib.request
import json
import urllib.error


def make_http_request(api_url, data=None, method='GET'):
    try:
        if data is not None:
            # Convert data to JSON string and then to bytes
            data = json.dumps(data).encode('utf-8')
        else:
            data = None

        # Define headers inside the function
        headers = {
            'Content-Type': 'application/json'
        }

        req = urllib.request.Request(api_url, data=data, headers=headers, method=method)

        with urllib.request.urlopen(req) as response:
            status_code = response.getcode()
            response_body = response.read()
            response_data = json.loads(response_body)
            return response_data, status_code

    except urllib.error.HTTPError as e:
        return {"error": e.reason}, e.code
    except urllib.error.URLError as e:
        return {"error": str(e)}, 500


def lambda_handler(event, context):
    api_url = "https://api.monobank.ua/bank/currency"

    json_data, status_code = make_http_request(api_url)

    return {
        "statusCode": status_code,
        "body": json_data
    }
